src/sonic-config-engine/minigraph.py

```diff
@@ -715,7 +736,26 @@ def parse_deviceinfo(meta, hwsku):
                 if desc != None:
                     port_descriptions[port_alias_map.get(alias, alias)] = desc.text
                 port_speeds[port_alias_map.get(alias, alias)] = speed
-    return port_speeds, port_descriptions
+
+            sysports = device_info.find(str(QName(ns, "SystemPorts")))
+            sys_ports = {}
+            if sysports is not None:
+                for sysport in sysports.findall(str(QName(ns, "SystemPort"))):
+                    slot_id = sysport.find(str(QName(ns, "SlotId")))
+                    portname = sysport.find(str(QName(ns, "Name"))).text
+                    device = sysport.find(str(QName(ns, "Device"))).text
+                    system_port_id = sysport.find(str(QName(ns, "SystemPortId"))).text
+                    switch_id = sysport.find(str(QName(ns, "SwitchId"))).text
+                    core_id = sysport.find(str(QName(ns, "CoreId"))).text
+                    core_port_id = sysport.find(str(QName(ns, "CorePortId"))).text
+                    speed = sysport.find(str(QName(ns, "Speed"))).text
+                    num_voq = sysport.find(str(QName(ns, "NumVoq"))).text
+                    key = "%s|%s" % (device, portname)
+                    if slot_id is not None:
+                       key = slot_id.text + '-' + key
```

Comment:
Need to use | as separator between all Slot, device and name. The suggested naming format is: 
Slot|Device|Name. 
Examples: 
```
multi_npu_platform_01|ASIC0|Cpu0
multi_npu_platform_01|ASIC0|Ethernet1/1
Slot1|Asic0|Cpu0
LineCard1|Asic0|Ethernet1
```